services/venice-controller/src/main/java/com/linkedin/venice/pushmonitor/AbstractPushMonitor.java

```diff
@@ -923,35 +934,77 @@ protected void checkWhetherToStartBufferReplayForHybrid(OfflinePushStatus offlin
-          LOGGER.info("Successfully {} for offlinePushStatus: {}", newStatusDetails, offlinePushStatus);
-        } catch (Exception e) {
-          // TODO: Figure out a better error handling...
-          String newStatusDetails = "Failed to kick off the buffer replay";
-          handleTerminalOfflinePushUpdate(offlinePushStatus, new ExecutionStatusWithDetails(ERROR, newStatusDetails));
-          LOGGER.error("{} for offlinePushStatus: {}", newStatusDetails, offlinePushStatus, e);
+          newStatusDetails.append("kicked off buffer replay");
+        } else if (!offlinePushStatus.getCurrentStatus().isTerminal()) {
+          LOGGER.info(
+              "{} is not ready to start buffer replay. Current state: {}",
+              offlinePushStatus.getKafkaTopic(),
+              offlinePushStatus.getCurrentStatus().toString());
         }
-      } else if (!offlinePushStatus.getCurrentStatus().isTerminal()) {
-        LOGGER.info(
-            "{} is not ready to start buffer replay. Current state: {}",
-            offlinePushStatus.getKafkaTopic(),
-            offlinePushStatus.getCurrentStatus().toString());
       }
+
+      if (isEOPReceivedInAllPartitions) {
+        // Check whether to send EOP for materialized view topic(s)
+        for (ViewConfig rawView: viewConfigMap.values()) {
+          VeniceView veniceView = ViewUtils.getVeniceView(
+              rawView.getViewClassName(),
+              new Properties(),
+              store.getName(),
+              rawView.getViewParameters());
+          if (veniceView instanceof MaterializedView) {
+            MaterializedView materializedView = (MaterializedView) veniceView;
+            for (String materializedViewTopicName: materializedView
+                .getTopicNamesAndConfigsForVersion(version.getNumber())
+                .keySet()) {
+              VeniceWriterOptions.Builder vwOptionsBuilder =
+                  new VeniceWriterOptions.Builder(materializedViewTopicName).setUseKafkaKeySerializer(true)
+                      .setPartitionCount(materializedView.getViewPartitionCount());
+              try (VeniceWriter veniceWriter = veniceWriterFactory.createVeniceWriter(vwOptionsBuilder.build())) {
+                veniceWriter.broadcastEndOfPush(Collections.emptyMap());
+              }
+              if (newStatusDetails.length() > 0) {
+                newStatusDetails.append(", ");
+              }
+              newStatusDetails.append("broadcast EOP to materialized view topic: ").append(materializedViewTopicName);
+            }
+          }
+        }
+        updatePushStatus(
+            offlinePushStatus,
+            ExecutionStatus.END_OF_PUSH_RECEIVED,
+            Optional.of(newStatusDetails.toString()));
+        LOGGER.info("Successfully {} for offlinePushStatus: {}", newStatusDetails.toString(), offlinePushStatus);
+      }
+    } catch (Exception e) {
+      String newStatusDetails = "Failed to start EOP procedures";
```

Comment:
nit pick: this string is constant. Maybe promote it into a private static variable?